opensafely/pull.py

```diff
@@ -62,6 +72,35 @@ def main(image, force):
         sys.exit(exc.stderr)
 
 
+def get_actions_from_project_file(project_yaml):
+    path = Path(project_yaml)
+    if not path.exists():
+        raise RuntimeError(f"Could not find {project_yaml} for preseed")
+
+    try:
+        with path.open() as f:
+            project = YAML(typ="safe", pure=True).load(path)
+    except (YAMLError, YAMLStreamError, YAMLWarning, YAMLFutureWarning) as e:
+        raise RuntimeError(f"Could not parse {project_yaml} for preseed: str(e)")
+
+    images = []
+    for action_name, action in project.get("actions", {}).items():
+        if not action:
+            continue
+        command = action.get("run", None)
+        if command is None:
+            continue
+
+        name, _, version = command.partition(":")
+        if name in IMAGES:
+            images.append(name)
+
+    if not images:
+        raise RuntimeError(f"No actions found in {project_yaml} to preseed")
+
+    return images
```

Comment:
I'm reminded that we really need to build a project.yaml parsing library!  Not something for this PR though.

Reply:
I'm reminded of that frequently!